addons/maintenance/models/maintenance.py

```diff
@@ -78,9 +78,50 @@ def _alias_get_creation_values(self):
         return values
 
 
+class MaintenanceMixin(models.AbstractModel):
+    _name = 'maintenance.mixin'
+    _check_company_auto = True
+    _description = 'Maintenance Maintained Item'
+
+    company_id = fields.Many2one('res.company', string='Company',
+        default=lambda self: self.env.company)
+    effective_date = fields.Date('Effective Date', default=fields.Date.context_today, required=True, help="This date will be used to compute the Mean Time Between Failure.")
+    maintenance_team_id = fields.Many2one('maintenance.team', string='Maintenance Team', compute='_compute_maintenance_team_id', store=True, readonly=False, check_company=True)
+    technician_user_id = fields.Many2one('res.users', string='Technician', tracking=True)
+    maintenance_ids = fields.One2many('maintenance.request')  # needs to be extended in order to specify inverse_name !
+    maintenance_count = fields.Integer(compute='_compute_maintenance_count', string="Maintenance Count", store=True)
+    maintenance_open_count = fields.Integer(compute='_compute_maintenance_count', string="Current Maintenance", store=True)
+    expected_mtbf = fields.Integer(string='Expected MTBF', help='Expected Mean Time Between Failure')
+    mtbf = fields.Integer(compute='_compute_maintenance_request', string='MTBF', help='Mean Time Between Failure, computed based on done corrective maintenances.')
+    mttr = fields.Integer(compute='_compute_maintenance_request', string='MTTR', help='Mean Time To Repair')
+    estimated_next_failure = fields.Date(compute='_compute_maintenance_request', string='Estimated time before next failure (in days)', help='Computed as Latest Failure Date + MTBF')
+    latest_failure_date = fields.Date(compute='_compute_maintenance_request', string='Latest Failure Date')
+
+    @api.depends('company_id')
+    def _compute_maintenance_team_id(self):
+        for record in self:
+            if record.maintenance_team_id.company_id and record.maintenance_team_id.company_id.id != record.company_id.id:
+                record.maintenance_team_id = False
+
+    @api.depends('effective_date', 'maintenance_ids.stage_id', 'maintenance_ids.close_date', 'maintenance_ids.request_date')
+    def _compute_maintenance_request(self):
+        for record in self:
+            maintenance_requests = record.maintenance_ids.filtered(lambda mr: mr.maintenance_type == 'corrective' and mr.stage_id.done)
+            record.mttr = len(maintenance_requests) and (sum(int((request.close_date - request.request_date).days) for request in maintenance_requests) / len(maintenance_requests)) or 0
+            record.latest_failure_date = len(maintenance_requests) and max(request.request_date for request in maintenance_requests) or False
```

Comment:
Can use `max`'s default to make the line shorter and simpler  (avoid `and`/`or`)
```suggestion
            record.latest_failure_date = max([request.request_date for request in maintenance_requests], default=False)
```

Reply:
fyi
pylint consider-using-generator : () rather than []
done